### ISSUE TITLE
Remind people to disable auto-update after applying the hack, permanently

### ISSUE THREAD

**Issue by martin-braun** [ddvk/remarkable-hacks#340]
People seem to forget to disable auto-update after applying the hack, permanently, thus losing the hacks, again. This leads to a high demand of new versions which causes higher activity in new issues and shorter ETAs.

Shorter ETAs introduce a higher chance of bugs, since a new version of the hack needs to be rushed to relief a higher amount of people. It's hard to live without something, when you are used it to it.

It would be better to print a warning in the console that reminds the user to disable auto-update again to reduce those incidents.

**Comment by StLa1** [comment]
Autoupdates sometimes re-enables itself

**Comment by Baccanno** [comment]
I do it the other way round, I check for hack update before updating. But your enhancement is on spot as I just disabled mine again after reading this !

**Comment by martin-braun** [comment]
@StLa1 can you be more specific here? I have WiFi enabled all the time and it never enabled itself.

@Baccanno You're welcome. There should be only one time gap in which you should enable auto-updates: When a new version has been hacked and no newer version is the around. I use internet all the time and auto-updates are off. 

If auto-updates enable itself I'd like to know when and how.

**Comment by StLa1** [comment]
I don't know the triggers. But I disabled it last year when started using ddvk. And a few months ago I enabled Wifi for the periodic sync. A few minutes later the banner saying "Restart to install update" appeared at the bottom. I check and auto-updates was enabled.

**Comment by martin-braun** [comment]
@StLa1 Maybe it's a bug, maybe you enabled it, accidentally. I found nothing on the internet, so let's see if someone can confirm this behavior or not.

**Comment by Trillianity** [comment]
> It's likely a bug, or intentional to improve adoption rates. But not an accidental gesture on my side.

I would like to confirm that. Early this year this happened to me too and auto-update reenabled itsself. Fortunately this happened only once early this year and I am not sure wether it is worth paying attention to it (as long as it does not happen more often).

**Comment by matael** [comment]
Same here, it got reenabled and I got updated to the latest version here. It's quite weird really and it could coincide with the moment I logged in the app again after months without using it.

**Comment by martin-braun** [comment]
@Trillianity @matael Thank you for your reports. That seriously fears me, now. Maybe @ddvk can spot this behavior in the binaries and patch it out. I also made a [Reddit shoutout](https://www.reddit.com/r/RemarkableTablet/comments/pfhh9v/people_report_remarkable_enables_autoupdate_by/) to seek some information.

**Comment by matael** [comment]
A friend reported yesterday that he was staying in 2.4 because he noticed the auto-updates seemed to reactivate from 2.5 and on.
That might be a starting point to investigate

**Comment by martin-braun** [comment]
@gerithebiker and @Blerkselmans report the same issue in #333 and I have a suspicion that it happens when you enable WiFi. My WiFi is always on, so I don't experience it, I'm not sure, though.

@matael This is a good starting point, indeed. A diff between 2.4 and 2.5 might help.

**Comment by Blerkselmans** [comment]
> @gerithebiker and @Blerkselmans report the same issue in #333 and I have a suspicion that it happens when you enable WiFi. My WiFi is always on, so I don't experience it, I'm not sure, though.

Not really, perhaps I was unclear. I have Wifi always enabled (I am possibly one of the few that can live without the ddvk hack for a week or two, although looking forward to the new release), but I had a double upgrade to 2.9 as well just like @gerithebiker.
Two weeks ago my Rm upgraded to 2.9 as anticipated and succesful, but last week it again told me to reboot because of an upgrade to 2.9. Assumed it was an additional patch. No problem for me, but if one loses the ability to downgrade (which is the problem of @gerithebiker).

**Comment by gerithebiker** [comment]
Now I am a bit confused... I tested on 2.9.1.217 if something triggers the automatic update to flip to "enabled". I tried both enabling/disabling flight mode - it did not change the auto-update status, and also, I did a reboot, and that did not change it either... 

So, was I just not careful enough and did not switch off autoupdate? I am not sure anymore...

**Comment by martin-braun** [comment]
hey @gerithebiker. @F5JSD commented on #333 that it happens for every reboot, but it does not for me, as well, I just tested it and I'm on 2.8.0.98. I start to suspect that they start to trick only some selected devices into update-mode to forcefully enroll their new version, but this is a real no-go for me. 

I want control over my device, screw this Apple mentality.

**Comment by StLa1** [comment]
Enabling WiFi is not the culprit. Nor is doing a restart. It's much rarer than that.

**Comment by matteodelabre** [comment]
For those who have encountered unwanted auto-updates reenabling (@StLa1, @Trillianity, @matael, @F5JSD), I wonder if running `systemctl mask --now update-engine` in a SSH session would prevent this from happening?

This command will prevent the update service from starting up. If you try it, you’ll see that you can’t even manually enable auto-updates from the settings menu. To reverse the effects, run `systemctl unmask update-engine`.

**Comment by martin-braun** [comment]
@matteodelabre That sounds promising, I made a new issue #352 to implement this to the installer. I suggest to discuss it further over there, because this issue is closed already.
